server/TelegramServerConfig.cpp

```diff
@@ -0,0 +1,107 @@
+#include "TelegramServerConfig.hpp"
+
+#include <QJsonDocument>
+#include <QJsonObject>
+#include <QJsonArray>
+#include <QJsonValue>
+#include <QFile>
+#include <QLoggingCategory>
+
+Q_LOGGING_CATEGORY(loggingCategoryConfig, "telegram.server.config", QtDebugMsg)
+
+namespace Telegram {
+
+namespace Server {
+
+Config::Config(const QString &fileName)
+{
+    if (fileName.isEmpty()) {
+        // m_fileName = QStringLiteral("/etc/TelegramServer/config.json");
+        m_fileName = QStringLiteral("config.json");
+    } else {
+        m_fileName = fileName;
+    }
+
+    // default config
+    m_serverConfiguration.dcOptions = {
+        Telegram::DcOption(QStringLiteral("127.0.0.1"), 11441, 1),
+        Telegram::DcOption(QStringLiteral("127.0.0.2"), 11442, 2),
+        Telegram::DcOption(QStringLiteral("127.0.0.3"), 11443, 3),
+    };
+    m_privateKeyFile = QStringLiteral("private_key.pem");
+}
+
+bool Config::load()
+{
+    QByteArray bytes;
+    QFile f(m_fileName);
+    if (!f.open(QIODevice::ReadOnly)) {
+        qCWarning(loggingCategoryConfig) << "Failed to open config file:" << m_fileName;
+        return false;
+    }
+    bytes = f.readAll();
+    f.close();
+
+    QJsonParseError jErr;
+    const QJsonDocument doc = QJsonDocument::fromJson(bytes, &jErr);
+    if (jErr.error != QJsonParseError::NoError) {
+        qCWarning(loggingCategoryConfig) << "Failed to parse config file: " << jErr.errorString();
+        qCWarning(loggingCategoryConfig) << "Using default config.";
+        return false;
+    }
+
+    const QJsonObject obj = doc.object();
+    if (obj.isEmpty()) {
+        qCWarning(loggingCategoryConfig) << "Failed to parse config file: " << jErr.errorString();
+        qCWarning(loggingCategoryConfig) << "Using default config.";
+        return false;
+    }
+
+    // read private key setting
+    m_privateKeyFile = obj.value(QLatin1String("privateKeyFile")).toString();
+
+    // read server addr/port config
+    m_serverConfiguration.dcOptions.clear();
+    const QJsonArray &jserverConfig = obj.value(QLatin1String("serverConfiguration")).toArray();
+    for (const QJsonValue &jval: jserverConfig) {
```

Comment:
Please keep spaces on both sides of ```:``` ([example in Qt](https://code.woboq.org/qt5/qtbase/src/corelib/tools/qstringlist.cpp.html#318))